File: Bots/JavaScript/Consumers/CodeFirst/WaterfallHostBot/skills/echoSkill.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

const { ActivityEx } = require('botbuilder-core');
const { SkillDefinition } = require('./skillDefinition');

class EchoSkill extends SkillDefinition {
  getActions () {
    return new Set(['Message']);
  }

  /**
   * @param {string} actionId
   */
  createBeginActivity (actionId) {
    if (!this.getActions().has(actionId)) {
      throw new Error(`[EchoSkill]: Unable to create begin activity for "${actionId}".`);
    }

    // We only support one activity for Echo so no further checks are needed
    const activity = ActivityEx.createMessageActivity();
    activity.name = actionId;
    activity.text = 'Begin the Echo Skill.';

    return activity;
  }
}

module.exports.EchoSkill = EchoSkill;

File: Bots/Python/Consumers/CodeFirst/WaterfallHostBot/skills/echo_skill.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from botbuilder.schema import Activity
from skills.skill_definition import SkillDefinition

SKILL_ACTION_MESSAGE = "Message"


class EchoSkill(SkillDefinition):
    def get_actions(self):
        return [SKILL_ACTION_MESSAGE]

    def create_begin_activity(self, action_id: str):
        activity = Activity.create_message_activity()

        if action_id == SKILL_ACTION_MESSAGE:
            activity.name = SKILL_ACTION_MESSAGE
            activity.text = "Begin the Echo Skill"

        else:
            raise Exception(f'Unable to create begin activity for "${action_id}"')

        return activity
